### ISSUE TITLE
Segfaults when fitting (or displaying?) R models when called from Julia - possible BLAS issue

### ISSUE THREAD

**Issue by dmbates** [JuliaInterop/RCall.jl#272]
```julia
  | | |_| | | | (_| |  |  Version 1.0.0 (2018-08-08)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |

julia> using RCall, LinearAlgebra

julia> BLAS._vendor
:mkl

julia> reval(R"names(Formaldehyde)")
RObject{StrSxp}
[1] "carb"   "optden"


julia> reval(R"lm(optden ~ 1 + carb, Formaldehyde)")

signal (11): Segmentation fault

signal (11): Segmentation fault
in expression starting at no file:0
in expression starting at no file:0
Segmentation fault (core dumped)
bates@thin206:/archive/Movies$ julia7
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0 (2018-08-08 06:46 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using RCall
[ Info: Recompiling stale cache file /home/bates/.julia/compiled/v0.7/RCall/8GFyb.ji for RCall [6f49c342-dc21-5d91-9882-a32aef131414]

julia> reval(R"lm(optden ~ carb, Formaldehyde)")
RObject{VecSxp}

Call:
lm(formula = optden ~ carb, data = Formaldehyde)

Coefficients:
(Intercept)         carb  
   0.005086     0.876286  
```
I have two versions of Julia installed - the v0.7.0 from the downloads site (the operating system is 64-bit Ubuntu 18.04) - and v1.0.0 built locally using MKL BLAS.  As can be seen, trying to fit a simple linear model in v1.0.0 through RCall results in a segfault whereas it works under v0.7.0.

My guess is that this is connected to the version of the BLAS/LAPACK.
```
bates@Latitude:/usr/lib/R/lib$ ldd libR.so
        linux-vdso.so.1 (0x00007ffe31f2e000)
        libblas.so.3 => /usr/lib/x86_64-linux-gnu/libblas.so.3 (0x00007f4171491000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f41710f3000)
        libreadline.so.7 => /lib/x86_64-linux-gnu/libreadline.so.7 (0x00007f4170eaa000)
        libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f4170c38000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f4170a12000)
        libbz2.so.1.0 => /lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007f4170802000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f41705e5000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f41703dd000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f41701d9000)
        libicuuc.so.60 => /usr/lib/x86_64-linux-gnu/libicuuc.so.60 (0x00007f416fe22000)
        libicui18n.so.60 => /usr/lib/x86_64-linux-gnu/libicui18n.so.60 (0x00007f416f981000)
        libgomp.so.1 => /usr/lib/x86_64-linux-gnu/libgomp.so.1 (0x00007f416f751000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f416f532000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f416f141000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f4172187000)
        libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f416ef17000)
        libicudata.so.60 => /usr/lib/x86_64-linux-gnu/libicudata.so.60 (0x00007f416d36e000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f416cfe0000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f416cdc8000)
```
Can anyone reproduce this on v1.0.0 with non-MKL BLAS?

**Comment by dmbates** [comment]
I think I have tracked it down a bit more.  On a 64-bit system R uses 32-bit integers in BLAS/LAPACK calls whereas Julia will use 64-bit integers.  Fortunately I am with "Mr. R", Martin Maechler, right now and he pointed out that you can find the version of BLAS/LAPACK in the sessionInfo output of R 3.5.0+
```
R> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.1 LTS

Matrix products: default
BLAS: /opt/intel/compilers_and_libraries_2018.2.199/linux/mkl/lib/intel64_lin/libmkl_rt.so
LAPACK: /opt/intel/compilers_and_libraries_2018.2.199/linux/mkl/lib/intel64_lin/libmkl_intel_lp64.so

locale:
 [1] LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                 
 [3] LC_TIME=en_US.UTF-8           LC_COLLATE=en_US.UTF-8       
 [5] LC_MONETARY=en_US.UTF-8       LC_MESSAGES=en_US.UTF-8      
 [7] LC_PAPER=en_US.UTF-8          LC_NAME=en_US.UTF-8          
 [9] LC_ADDRESS=en_US.UTF-8        LC_TELEPHONE=en_US.UTF-8     
[11] LC_MEASUREMENT=en_US.UTF-8    LC_IDENTIFICATION=en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lme4_1.1-18-1 Matrix_1.2-14

loaded via a namespace (and not attached):
[1] minqa_1.2.4     MASS_7.3-50     compiler_3.5.1  Rcpp_0.12.18   
[5] splines_3.5.1   nlme_3.1-137    grid_3.5.1      nloptr_1.0.4   
[9] lattice_0.20-35
```

It turns out that libR.so is loads a BLAS dll when it is loaded but it isn't until R starts running and a call is made to an LAPACK function that an LAPACK dll is opened.  It looks as if when R is running from within Julia it can use the MKL LAPACK that Julia already has open, resulting in a 64-bit/32-bit conflict.

**Comment by dmbates** [comment]
Yes, I do use the official Julia binaries.  I was comparing some linear-algebra-intensive fits in lme4 and MixedModels and, for fairness, I really should have them both use the same BLAS.  On this machine the default BLAS is MKL.  I eventually used an official binary to work on RCall and my locally compiled version to do the testing in MixedModels.

I just raised this issue in case it comes up another time.  R doesn't seem to be as discerning as it should be about integer size when it opens an LAPACK dll.

I seem to remember that it is possible to compile julia with MKL BLAS specifying 32-bit integers on a 64-bit system.  Effectively all the calls to LAPACK and BLAS truncate their integer arguments which are, in most cases, scalars and only occasionally vectors that will become pivot vectors, etc., That might be another approach to try.

**Comment by palday** [comment]
@dmbates You fixed this by setting some environment variable, if I remember correctly. Care to post a canonical solution, pin and close the issue so that the problem and solution are easy to find? :smile:

**Comment by dmbates** [comment]
@palday  BLAS wasn't quite the issue that I needed to work around recently.  What was happening to me recently is that compiling the lme4 package for R caused it to link to the system's version of `libstdc++.so` whereas Julia comes with its own, slightly older version of `libstdc++.so` in
`joinpath(Sys.BINDIR, "..", "lib", "julia")`

When you start R through RCall then 
`R"require('lme4')"`
the older version of `libstdc++.so` is the first one on the dynamic load path.  The way that I got around this problem is to set `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6` before starting julia.

It is tedious but I don't see a way around it.
